ttnn/cpp/ttnn/operations/moreh/moreh_norm/device/moreh_norm_w/moreh_norm_w.cpp

```diff
@@ -0,0 +1,215 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <functional>
+#include <map>
+#include <optional>
+#include <utility>
+#include <vector>
+
+#include "ttnn/run_operation.hpp"
+#include "ttnn/tensor/tensor.hpp"
+#include "ttnn/tensor/tensor_impl.hpp"
+#include "ttnn/deprecated/tt_dnn/op_library/moreh_helper_functions.hpp"
+#include "ttnn/deprecated/tt_dnn/op_library/moreh_norm/moreh_norm_op.hpp"
+#include "tt_metal/common/work_split.hpp"
+#include "tt_metal/detail/util.hpp"
+#include "tt_metal/host_api.hpp"
+
+namespace tt {
+
+namespace operations {
+
+namespace primary {
```

Comment:
```suggestion
namespace ttnn::operations::moreh::moreh_norm {
```
In ttnn, the primary namespace is not used.

Reply:
I did not, as the moreh_norm_backward has not been reallocated. I'm working on that (backward processing).

Reply:
Oh you're right, I'll delete it now.